composables/useCharacterDetails.ts

```diff
@@ -0,0 +1,45 @@
+import { gql } from "@apollo/client/core";
+import { useQuery } from "@vue/apollo-composable";
+import { ID } from "graphql-ws";
+import { CharacterQuery, CharacterQueryVariables } from "~/types/graphql/generated";
+
+const CharacterQueryDefinition = gql`
+    query character($id: ID!) {
+        character(
+            id: $id
+        ) {
+            id
+            name
+            status
+            species
+            type
+            gender
+            origin {
+                name
+            }
+            location {
+                name
+            }
+            image
+            episode {
+                id
+                name
+                episode
+                air_date
+                created
+                characters {
+                    id
+                    name
+                }
+            }
+        }
+    }
+`;
+export function useCharacterDetails (props: { id?: ID }) {
```

Comment:
I guess id prop of the hook should be `string` or `number`, TS doesn't recognize ID as a type